CommitsReporter: add support for per-commit-view

yamidark opened the pull request:
Part of #134, #135 
```
The dashboard only supports daily and weekly interval views for the
commits.

This is due to commits.json file only holding the daily and weekly
interval contributions summary data.

As part of implementing a per-commit-view on the dahsboard, let's
update commits.json to store the list of each CommitResult inside
the daily interval contribution data. This list of CommitResult with the
full commit hash will also allow for tracking of previously analyzed
commits for incremental build.

As the weekly interval view is now generated on the frontend based
on the daily view by #129, there is also no longer a need to store the
weekly interval contribution data in commits.json.

Let's remove the weekly interval contribution data in commits.json
to reduce the size of the json files.
```
Note: since the frontend v2 is still not merged in and this PR is branched of master, 
the dashboard currently cannot be seen.

yamidark commented:
@ongspxm @AdityaA1998 @yong24s @eugenepeh For your reviews, since this is necessary for our incremental build.

yong24s commented on src/main/java/reposense/report/CommitReportJson.java:
```diff
@@ -6,20 +6,21 @@
 import reposense.model.Author;
 
 /**
  * Class that holds the data to be serialized into JSON format in `commits.json`.
```
You may want to also update the class header so it is consistent with our code style.

```Stores the data to be serialized...```

yamidark replied:
As this is a class header and not a method header, I don't think there's a need to adjust the header to start with a verb.

yamidark commented:
As suggested by @ongspxm, I will be updating this to be PR to store the commits in an array of dates, with each date containing an array of commits for that day, so can hold back the reviews till then.

yamidark commented:
Changes made:
Renamed `AuthorIntervalContribution` to `AuthorDailyContribution`.
Update `AuthorDailyContribution` to also store the author's commit made for that day, effectively mixing `authorDailyIntervalContribution` and list of `commitResults` together.
Removed `authorDailyIntervalContribution` and `authorWeeklyIntervalContribution`.

yamidark commented:
@ongspxm For your review 1st, check the updated format of the `commits.json` if its correct.

yamidark commented:
@eugenepeh Will need to merge this into another branch 1st instead of master, as the dashboard won't be generated in this PR due to changing of `commits.json`, to be together with #174. Do we continue using the `frontend` branch?

eugenepeh commented:
@yamidark I thought I have created an incremental branch as you guys have requested?

yamidark commented:
@eugenepeh For your review.

eugenepeh commented:
weird that 1f222c0 shows up in a few PR. Can you rebase first? So that I can ignore the changes made by 1f222c0.

yamidark commented:
@eugenepeh Reminder for your review.

eugenepeh commented on src/main/java/reposense/commits/model/AuthorDailyContribution.java:
```diff
@@ -0,0 +1,55 @@
+import java.util.List;
+
+
+public class AuthorDailyContribution {
```
Please add the header comments

yamidark commented:
>Which PR is it that needs to go in together with this?

@eugenepeh #174,
@ongspxm Please test on your side from this branch if the dashboard works properly, and with the new per-commit view.

eugenepeh commented:
@yamidark you can merge it once you update per-commit-view to be in sync with master.
